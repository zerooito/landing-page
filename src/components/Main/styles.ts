import styled from 'styled-components'

export const Wrapper = styled.main`
  background-color: #06092b;
  height: 100%;
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`

export const Title = styled.h1`
  font-size: 2rem;
  color: #d83a3a;
  margin-bottom: 1rem;
`
