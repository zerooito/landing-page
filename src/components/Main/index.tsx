import * as S from './styles'

const Main = ({ title = 'Boilerplatezada.' }) => {
  return (
    <S.Wrapper>
      <S.Title> {title} </S.Title>
    </S.Wrapper>
  )
}

export default Main
