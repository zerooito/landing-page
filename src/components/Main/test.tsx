import { screen, render } from '@testing-library/react'
import Main from '.'

describe('<Main />', () => {
  it('shoulder render the heading', () => {
    const { container } = render(<Main />)
    expect(
      screen.getByRole('heading', { name: /boilerplatezada./i })
    ).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the colors correctly', () => {
    const { container } = render(<Main />)
    expect(container.firstChild).toHaveStyle({ 'background-color': '#06092b' })
  })
})
