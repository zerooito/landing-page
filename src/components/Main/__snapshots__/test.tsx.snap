// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> shoulder render the heading 1`] = `
.c0 {
  background-color: #06092b;
  height: 100%;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  font-size: 2rem;
  color: #d83a3a;
  margin-bottom: 1rem;
}

<main
  class="c0"
>
  <h1
    class="c1"
  >
     
    Boilerplatezada.
     
  </h1>
</main>
`
